cmd/soroban-cli/src/commands/keys/address.rs

```diff
@@ -45,6 +50,11 @@ impl Cmd {
     pub fn public_key(&self) -> Result<stellar_strkey::ed25519::PublicKey, Error> {
         if let Ok(key) = stellar_strkey::ed25519::PublicKey::from_string(&self.name) {
             Ok(key)
+        } else if let Ok(unresolved) = self.name.parse::<UnresolvedMuxedAccount>() {
```

Comment:
@willemneal When I was testing after merging main, I realized I needed to change this function. I added this `else if` branch to get the public key from the secure store without first getting the private key. Can you take a look and make sure this looks correct? I tested it locally but would love another set of eyes on it.